src/deploy/kubernetes_functions.js

```diff
@@ -1,260 +1,292 @@
+            console.log('waiting for exeternal ips to be allocated. may take some time..');
+            // 20 minutes timeout by default
+            return P.resolve()
+                .then(async () => {
+                    const delay = 10000;
+                    let address = null;
+                    // get external ip
+                    while (!address) {
+                        service = await this.kubectl_get('service', service_name);
+                        const ingress = _.get(service, 'status.loadBalancer.ingress.0.ip');
+                        const hostname = _.get(service, 'status.loadBalancer.ingress.0.hostname');
+                        const external_ip = _.get(service, 'spec.externalIPs.0');
+                        address = ingress || hostname || external_ip;
+                        if (!address) {
+                            await P.delay(delay);
+                        }
+                    }
+                    return { address, ports };
+                })
+                .timeout(timeout);
         }
-        //TODO: check and print when it delete
     }
 
-    async upgrade_server() {
-        //TODO::
-        // delete server Container
-        // create new server container
+
+
+
+    async wait_for_pod_ready(pod_name, additional_test, options = {}) {
+        const delay = 10000;
+        console.log(`waiting for pod ${pod_name} to become ready..`);
+        // 20 minutes timeout by default
+        const { timeout = 10 * 60000 } = options;
+        return P.resolve()
+            .then(async () => {
+                let ready = false;
+                while (!ready) {
+                    try {
+                        const pod = await this.kubectl_get('pod', pod_name);
+                        ready = (pod.status.containerStatuses[0].ready);
+                        if (additional_test) {
+                            ready = ready && additional_test();
+                        }
+                        if (!ready) {
+                            throw new Error('not ready');
+                        }
+                    } catch (err) {
+                        await P.delay(delay);
+                    }
+                }
+                console.log(`pod ${pod_name} is ready`);
+            })
+            .timeout(timeout);
+    }
+
+    async test_http_req(test_url, expected_status, timeout) {
+        try {
+            const req_options = {
+                url: test_url,
+                timeout: timeout || 10000,
+                rejectUnauthorized: false,
+            };
+            const res = await P.fromCallback(callback => request(req_options, callback));
+            if (res.statusCode !== expected_status) {
+                return false;
+            }
+            return true;
+        } catch (err) {
+            return false;
+        }
     }
 
 }
 
-module.exports = KubernetesFunctions;
+KubernetesFunctions.IS_IN_POD = process.env.CONTAINER_PLATFORM === 'KUBERNETES';
```

Comment:
We are already checking the condition here:
https://github.com/noobaa/noobaa-core/pull/5559/files#diff-81f6762b40dee5e23e074a3bf0d4ed99R17

Reply:
will remove